shared/fn-ref-ex.asciidoc

```diff
@@ -0,0 +1,83 @@
+////
+Below is a sample function reference listing.
+A heading and short description are visible.
+Other content is hidden in a collapsible block.
+////
+
+[[sample-fn]]
+=== `substring`
+
+////
+A brief 1-2 sentence description.
+////
+Does a cool thing.
+
+[%collapsible]
+====
+
+////
+An example using the function.
+The goal is to show an input and output.
+If needed, you can include the return output in a comment.
+////
+
+*Example*
+[source,txt]
+----
+substring("quick brown fox", 0, 5)      // returns "quick"
+substring("quick brown fox", 6, 11)     // returns "brown"
+substring("quick brown fox", 6)         // returns "b"
+substring("quick brown fox", -3, -1)    // returns "fo"
+substring("quick brown fox", -3)        // returns "f"
+----
+
+////
+A snippet outlining the function syntax.
+Unnamed parameters are included using angle brackets (e.g. `<parm>`).
+Optional parameters are included using square brackets (e.g. [<parm>]).
+////
+
+*Syntax*
+[source,txt]
+----
+substring(<source>, <start_pos>[, <end_pos>])
+----
+
+////
+Guidelines for parameter documentation
+***************************************
+* Use a definition list.
+* End each definition with a period.
+* Include whether the parameter is Optional or Required and the data type.
+* A † denotes a parameter can be passed multiple times.
```

Comment:
> A † denotes a parameter can be passed multiple times.

I actually had to hunt for that legend in the Kibana example, since I missed see that at the top.  Would it be feasible to have these as callouts or footnotes instead so they're linked?

Reply:
I think that's a great idea. I added this as a footnote with ec18644ef072354dbeecb91b838d9a9f80040996. Let me know if you have any suggestions on the implementation.